api/envoy/config/bootstrap/v3/bootstrap.proto

```diff
@@ -186,6 +198,29 @@ message Bootstrap {
   // Specifies optional bootstrap extensions to be instantiated at startup time.
   // Each item contains extension specific configuration.
   repeated core.v3.TypedExtensionConfig bootstrap_extensions = 21;
+
+  // Configuration sources that will participate in
+  // *udpa.core.v1.ResourceLocator* authority resolution. The algorithm is as
+  // follows:
+  // 1. The authority field is taken from the *udpa.core.v1.ResourceLocator*, call
+  //    this *resource_authority*.
+  // 2. *resource_authority* is compared against the authorities in any peer
+  //    *ConfigSource*. The peer *ConfigSource* is the configuration source
+  //    message which would have been used unconditionally for resolution
+  //    with opaque resource names. If there is a match with an authority, the
+  //    peer *ConfigSource* message is used.
+  // 3. *resource_authority* is compared sequentially with the authorities in
+  //    each configuration source in *config_sources*. The first *ConfigSource*
+  //    to match wins.
+  // 4. As a fallback, if no configuration source matches, then
+  //    *default_configuration_source* is used.
+  // [#not-implemented-hide:]
+  repeated core.v3.ConfigSource config_sources = 22;
+
+  // Default configuration source for *udpa.core.v1.ResourceLocator* if all
+  // other resolution fails.
+  // [#not-implemented-hide:]
+  core.v3.ConfigSource default_config_source = 23;
```

Comment:
Would this end up being required in v4? If not required, can we ultimately document what the behavior is if there is no matching config source for an authority?

Reply:
I think if none is specified, we can say resolution fails. I think ultimately we might want to consider some kind of fall-back DNS resolution as default, but that will require some thought.